syntheseus/search/node_evaluation/common.py

```diff
@@ -37,15 +38,17 @@ def _evaluate_nodes(self, nodes, graph=None) -> Sequence[float]:
         return [-v for v in super()._evaluate_nodes(nodes, graph)]
 
 
-class ReactionModelProbPolicy(ReactionModelBasedEvaluator[MolSetNode]):
-    """Evaluator that uses the reactions' probability to form a policy (useful for OR-MCTS)."""
+class ReactionModelProbPolicy(ReactionModelBasedEvaluator[Union[MolSetNode, AndNode]]):
```

Comment:
Do you think it is better to put this into a single class with a node type check instead of having two separate classes?

Reply:
Are you suggesting that I replace `Union[MolSetNode, AndNode]` with a single `BaseGraphNode` for typing? Or did you mean something else? I am using a single ReactionModelProbPolicy class to handle both MolSetNode in OR-MCTS and AndNode in PDVN-MCTS here

Reply:
In the current branch these are [two separate classes](https://github.com/microsoft/syntheseus/blob/8516d3108070f97a3e31f185fb6c2e3c38ff8d74/syntheseus/search/node_evaluation/common.py#L27-L51), so my question is why merge them into one class?

Reply:
The other class on `main` is negative log cost, whereas here we use probabilities. Would you suggest to now have _three_ classes for each of the usecases?

Reply:
Ah, you are right (I didn't read this carefully). Let's just merge them then as Guoqing suggests.